src/linear_algebra/quadratic_form.lean

```diff
@@ -938,3 +938,100 @@ let ⟨w, ⟨hw₁⟩⟩ := Q.equivalent_weighted_sum_squares_of_nondegenerate'
 end complex
 
 end quadratic_form
+
+namespace units
+
+/-- The sign function that maps negative real numbers to -1 and nonnegative numbers to 1. -/
+noncomputable def sign (r : units ℝ) : ℝ := if (r : ℝ) < 0 then -1 else 1
```

Comment:
I think this definition (and the four lemmas below it) should move to `data.real.basic` or something like that.
(Maybe a new file `data.real.sign` is even better.)

Reply:
Is this now `units.sign`? That sounds like it's maybe a bit too generic a name (someone might want to do the same thing for rationals at some point).
From a Lean POV, I think I would define `real.sign` and just define it on all real numbers. (Sending `0` either to `0` or to `1`).

Reply:
I've added it to `data.real.sign` now